arm-iothub/2017-01-19/swagger/iothub.json

```diff
@@ -0,0 +1,1962 @@
+          "type": "string"
+        },
+        "subscriptionId": {
+          "description": "The subscription identifier of the event hub endpoint.",
+          "type": "string"
+        },
+        "resourceGroup": {
+          "description": "The name of the resource group of the event hub endpoint.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "name",
+        "connectionString"
+      ]      
+    },    
+    "RouteProperties": {
+      "description": "The properties related to a routing rule based on which the IoT hub routes messages to endpoints.",      
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the route. name can only include alphanumeric characters, periods, underscores, hyphens with maximum length of 64 characters and must be unique.",      
+          "type": "string",
+          "pattern": "^[A-Za-z0-9-._]{1,64}$"
+        },
+        "source": {
+          "description": "The source to which the routing rule is to be applied to. e.g. DeviceMessages",      
+          "enum": ["DeviceMessages"],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "RoutingSource",
+            "modelAsString": true
+          }
+        },
+        "condition": {
+          "description": "The condition which is evaluated in order to apply the routing rule. If the condition is not provided it will evaluate to true by default. For grammar, See: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-query-language",      
+          "type": "string"
+        },
+        "endpointNames": {
+          "description": "The list of endpoints to which the messages that satisfy the condition are routed to. Currently only 1 endpoint is allowed.",
+          "minItems": 1,      
+          "maxItems": 1,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "isEnabled": {
+          "description": "Used to specify whether a route is enabled or not.",
+          "type": "boolean"
+        }
+      },
+      "required": [
+        "name",
+        "endpointNames",
+        "source",
+        "isEnabled"
+      ]
+    },
+    "IotHubDescription": {
+      "description": "The description of the IoT hub.",
+      "type": "object",
+      "properties": {
+        "subscriptionid": {
+          "description": "The subscription identifier.",
+          "type": "string"
+        },
+        "resourcegroup": {
+          "description": "The name of the resource group that contains the IoT hub. A resource group name uniquely identifies the resource group within the subscription.",
+          "type": "string"
+        },
+        "etag": {
+          "description": "The Etag field is *not* required. If it is provided in the response body, it must also be provided as a header per the normal ETag convention.",
+          "type": "string"
+        },
+        "properties": {
```

Comment:
No strong concerns but this will definitely provide better developer work-flow from SDK perspective. It's mostly a recommendation at this point but in future if you decide to apply `x-ms-client-flatten` then that'll be a breaking change in SDK. Considering you are revising version this time, I think you have opportunity here if there already some breaking changes going in with this swagger.